systemd-hosts.d.service

```diff
@@ -3,13 +3,66 @@ Description=Keep your hosts records in logically separated files
 After=local-fs.target
 
 [Service]
+# Security
+ProtectSystem=yes
+ProtectHome=yes
+PrivateDevices=yes
+ProtectKernelTunables=yes
+ProtectKernelModules=yes
+ProtectControlGroups=yes
+ProtectKernelLogs=yes
+SystemCallFilter=@system-service
+SystemCallErrorNumber=EPERM
+NoNewPrivileges=yes
+PrivateTmp=yes
+LockPersonality=yes
+MemoryDenyWriteExecute=yes
+ProtectClock=yes
+RestrictNamespaces=yes
+RestrictRealtime=yes
+RestrictSUIDSGID=yes
+RuntimeDirectory=systemd/hosts.d
+RuntimeDirectoryPreserve=yes
```

Comment:
Normally I'd be all-in for hardening while packaging anything remotely non-trivial. But at this case, everything we do here is fairly trivial (if you discovered this on AUR, you probably already had to look through sources yourself). So, in this context, I'd put emphasis on readability/auditability right now, rather than creating an opaque wall of "Security" guards.

Btw, going this route, from a quick look, `SystemCallFilter` can be narrowed to only fs operation, and `{Dynamic}User` is unset, which overall, I assume, leaves process with uid 0, which, by itself, _may_ expose some additional privileged escalation routes.

Reply:
> Normally I'd be all-in for hardening while packaging anything remotely non-trivial. But at this case, everything we do here is fairly trivial (if you discovered this on AUR, you probably already had to look through sources yourself). So, in this context, I'd put emphasis on readability/auditability right now, rather than creating an opaque wall of "Security" guards.

I think it's better to be in habit to harden the chain wherever possible. I'd vote for additional comments over omission.

> Btw, going this route, from a quick look, `SystemCallFilter` can be narrowed to only fs operation, and `{Dynamic}User` is unset, which overall, I assume, leaves process with uid 0, which, by itself, _may_ expose some additional privileged escalation routes.

Good catch, I'll try tightening it.